vcd/resource_vcd_org_vdc_access_control.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"strings"
+)
+
+func resourceVcdOrgVdcAccessControl() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVdcVdcAccessControlCreateUpdate,
+		ReadContext:   resourceVdcVdcAccessControlRead,
+		UpdateContext: resourceVdcVdcAccessControlCreateUpdate,
+		DeleteContext: resourceVdcVdcAccessControlDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVdcVdcAccessControlImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"shared_with_everyone": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether the VDC is shared with everyone",
+			},
+			"everyone_access_level": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+				Description:  "Access level when the VDC is shared with everyone (only ReadOnly is available). Required when shared_with_everyone is set",
+			},
+			"shared_with": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"user_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "ID of the user to which we are sharing. Required if group_id is not set",
+						},
+						"group_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "ID of the group to which we are sharing. Required if user_id is not set",
+						},
+						"subject_name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Name of the subject (group or user) with which we are sharing",
+						},
+						"access_level": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice([]string{types.ControlAccessReadOnly}, true),
+							Description:  "The access level for the user or group to which we are sharing. (Only ReadOnly is available)",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVdcVdcAccessControlCreateUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	var accessControl types.ControlAccessParams
+
+	isSharedWithEveryone := d.Get("shared_with_everyone").(bool)
+	everyoneAccessLevel := d.Get("everyone_access_level").(string)
+	sharedList := d.Get("shared_with").(*schema.Set).List()
+
```

Comment:
what happens if I have VDC and access controls? creating the with VDC will override existing ones?

Reply:
Ok, this is interesting.
So turns out now the resource allow to do two things. User might be working with a VDC before hand, then the VDC will be get by name from either the provider block or resource block. If the VDC is created from Terraform, there's a new schema attribute called `vdc_id` which allows to pass the ID from the VDC created from Terraform. That parameter **MUST** be used only when creating VDC from Terraform. Must be left blank otherwise.

Reply:
And that parameter still need to be documented.

Reply:
> If the VDC is created from Terraform, there's a new schema attribute called vdc_id which allows to pass the ID from the VDC created from Terraform. That parameter MUST be used only when creating VDC from Terraform. Must be left blank otherwise.

What additional benefit would it give? Wouldn't one be able to use `vdc` field as well?

Reply:
Let's back to the original question: I have VDC and use UI or other means to create access controls.
What happens when create resource and add set new access controls?